services/web/server/tests/unit/with_dbs/04/workspaces/test_workspaces__moving_folders_between_workspaces.py

```diff
@@ -0,0 +1,272 @@
+    )
+    resp = await client.put(f"{url}")
+    await assert_status(resp, status.HTTP_204_NO_CONTENT)
+    # add second project to the folder
+    url = client.app.router["replace_project_folder"].url_for(
+        folder_id=f"{second_folder['folderId']}", project_id=f"{second_project['uuid']}"
+    )
+    resp = await client.put(f"{url}")
+    await assert_status(resp, status.HTTP_204_NO_CONTENT)
+
+    ## Double check whether everything is setup OK
+    url = (
+        client.app.router["list_projects"]
+        .url_for()
+        .with_query({"folder_id": f"{second_folder['folderId']}"})
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 2
+
+    url = (
+        client.app.router["list_projects"]
+        .url_for()
+        .with_query({"folder_id": f"{first_folder['folderId']}"})
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 0
+
+    url = client.app.router["list_projects"].url_for().with_query({"folder_id": "null"})
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 0
+
+    url = client.app.router["list_folders"].url_for().with_query({"folder_id": "null"})
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+    url = (
+        client.app.router["list_folders"]
+        .url_for()
+        .with_query({"folder_id": f"{first_folder['folderId']}"})
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+    return f"{second_folder['folderId']}"
+
+
+async def _move_folder_to_workspace_and_assert(
+    client: TestClient, folder_id: str, workspace_id: str
+):
+    assert client.app
+
+    # MOVE
+    base_url = client.app.router["replace_folder_workspace"].url_for(
+        folder_id=folder_id,
+        workspace_id=workspace_id,
+    )
+    resp = await client.put(f"{base_url}")
+    await assert_status(resp, NO_CONTENT)
+
+    # ASSERT
+    url = (
+        client.app.router["list_projects"]
+        .url_for()
+        .with_query(
+            {
+                "folder_id": folder_id,
+                "workspace_id": workspace_id,
+            }
+        )
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 2
+
+    url = (
+        client.app.router["list_folders"]
+        .url_for()
+        .with_query(
+            {
+                "folder_id": folder_id,
+                "workspace_id": workspace_id,
+            }
+        )
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+
+@pytest.mark.parametrize("user_role,expected", [(UserRole.USER, status.HTTP_200_OK)])
```

Comment:
we are copying this parametrization everywhere when there is no need. If you only want to have a `logged_user`with `user_role=USER`, just create a fixture `user_role` in the scope you want, e.g. here or in a  `conftest.py`

```python
# in conftest.py or somewhere on the top one time!
@pytest.fixture
def user_role() -> UserRole:
     return UserRole.USER

# all your tests already assume a normal user 
     
async def test_moving_between_private_and_shared_workspaces(
    client: TestClient,
    logged_user: UserInfoDict,
    mock_catalog_api_get_services_for_user_in_product: MockerFixture,
    fake_project: ProjectDict,
    workspaces_clean_db: None,
):
     # just to be sure
      logged_user["role"] == UserRole.USER